darwin/DarwinProcess.c

```diff
@@ -327,6 +320,25 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
       }
    }
 
+   if (proc->tty_name == NULL && (dev_t)proc->tty_nr != NODEV) {
+      /* The call to devname() is extremely expensive (due to lstat)
+       * and represents ~95% of htop's CPU usage when there is high
+       * process turnover.
+       *
+       * To mitigate this we only fetch TTY information if the TTY
+       * field is enabled in the settings.
+       */
+      if (Settings_hasField(proc->settings, TTY)) {
+         const char *name = devname(proc->tty_nr, S_IFCHR);
+         if (name) {
+            proc->tty_name = xStrndup(name, 512);
+         } else {
+            /* devname failed: prevent us from calling it again */
+            proc->tty_nr = NODEV;
+         }
+      }
+   }
```

Comment:
Rationale good, but use the accumulated `flags` for checking instead. Compare with e.g. how the `CWD` column is handled just above this block of code.

Reply:
Good point I totally missed the accumulated `flags`. Fixed and I also removed the `Settings_hasField` and `ProcessField_hasField` functions.